pkg/arvo/lib/hood/drum.hoon

```diff
@@ -329,23 +330,65 @@
   [%give %fact ~[/drum] %dill-blit !>(dill-blit)]
 ::
 ++  se-adit                                           ::  update servers
-  ^+  .
-  ::  ensure dojo connects after talk
-  =*  dojo-on-top  |=([a=* b=*] |(=(%dojo a) &(!=(%dojo b) (aor a b))))
-  %+  roll  (sort ~(tap in ray) dojo-on-top)
-  =<  .(con +>)
-  |:  $:{wel/well:gall con/_..se-adit}  ^+  con
-  =.  +>.$  con
-  =+  hig=(~(get by fur) q.wel)
-  ?:  &(?=(^ hig) |(?=(~ u.hig) =(p.wel syd.u.u.hig)))  +>.$
-  =.  +>.$  (se-text "activated app {(trip p.wel)}/{(trip q.wel)}")
-  %-  se-emit(fur (~(put by fur) q.wel ~))
-  =/  =wire  [%drum p.wel q.wel ~]
-  [%pass wire %arvo %g %conf [our.hid q.wel] our.hid p.wel]
+  ^+  this
+  |^
+  %+  reel
+    (sort ~(tap in ray) sort-by-priorities)
+  |=  [=well:gall that=_this]
+  ^+  that
+  =/  =wire  [%drum p.well q.well ~]
+  =/  hig=(unit (unit server))
+    (~(get by fur) q.well)
+  ?:  &(?=(^ hig) |(?=(~ u.hig) =(p.well syd.u.u.hig)))
+    that
+  =.  fur
+    (~(put by fur) q.well ~)
+  =.  that
+    (se-text "activated app {(trip p.well)}/{(trip q.well)}")
+  %-  se-emit
+  [%pass wire %arvo %g %conf [our.hid q.well] our.hid p.well]
```

Comment:
Hmm, somehow this comment got eaten.

This doesn't preserve the changes in the accumulator because we never read from `that`.  In particular, `se-text` and `se-emit` add moves to our state, but they get ignored the next iteration of `reel`.  You can resolve this by adding `=.  this  that` after the `|=`, or by making sure all functions are called like `se-text:that`, or by unrolling to a manual `|-`.

This bug will present as "works to start one app but not multiple apps", which is the kind of bug that may not be found until it's on livenet and is often hard to pinpoint.  I really hate bugs where I *know* I did the right thing, just the moves/state changes got silently dropped somewhere else in the code.  Personally, I shy away from ever using `this` as the accumulator in `reel`/`roll` for exactly this reason.

Reply:
So your preference would be for recursion here?

Reply:
Yeah, I would just do a `|-` here.